docs/src/pages/guides/typescript/typescript.md

```diff
@@ -256,3 +256,20 @@ example concerning `Button` and `react-router-dom` checkout [this Button demo](/
 Not every component fully supports any component type you pass in. If you encounter a
 component that rejects its `component` props in TypeScript please open an issue.
 There is an ongoing effort to fix this by making component props generic.
+
+## Handling `value` and event handlers
+
+Many components concerned with user input offer a `value` prop or event handlers
+which include the current `value`. In most situations that `value` is only handled
+within React which allows it be of any type like objects or arrays.
+
+However, that type cannot be verified at compile time in situations where it depends
+on the components children e.g. for `Select` or `RadioGroup`. This means that
+the soundest option is to type it as `unknown` and let the developer decide
```

Comment:
Should we do the same for the Slider?
https://github.com/mui-org/material-ui/blob/e499a0e93f04628724332d42c8156232786cc569/docs/src/pages/components/slider/ContinuousSlider.tsx#L19

Reply:
I'll take a look but I don't think so. As long as the type only depends on the props of a single components is fine. It's only problematic if the type is passed to children/parents because typescript can't check those relationships.